clang/include/clang/CConv/ProgramInfo.h

```diff
@@ -69,16 +70,17 @@ class ProgramInfo : public ProgramVariableAdder {
   // should all be empty. 
   void exitCompilationUnit();
 
-  CVarSet
-      &getPersistentConstraintVars(Expr *E, ASTContext *AstContext);
+  CVarSet &getPersistentConstraintVars(Expr *E, ASTContext *AstContext);
   // Get constraint variable for the provided Decl
-  CVarSet getVariable(clang::Decl *D,
-                                             clang::ASTContext *C);
+  CVarSet getVariable(clang::Decl *D, clang::ASTContext *C);
 
   // Retrieve a function's constraints by decl, or by name; nullptr if not found
-  std::set<FVConstraint *> *getFuncConstraints(FunctionDecl *D, ASTContext *C);
-  std::set<FVConstraint *> *getExtFuncDefnConstraintSet(std::string FuncName);
-  std::set<FVConstraint *> *getStaticFuncConstraintSet(std::string FuncName, std::string FileName);
+  const std::set<FVConstraint *> *getFuncConstraints
+      (FunctionDecl *D, ASTContext *C) const;
```

Comment:
What does a `const` return type mean? What does `const` after the prototype mean?

Reply:
`const` after prototype says that `this` is not mutated. `const` return means that the return value *cannot* be mutated by the caller.